src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onToggle: () => void | Promise<void>;
+  isLoading?: boolean;
+  onDeleted: (id: number) => void;
+  onEditSubmit: (
+    id: number,
+    oldTitle: string,
+    newTitle: string,
+  ) => void | Promise<void>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onToggle,
+  isLoading,
+  onDeleted,
+  onEditSubmit,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editingTitle, setEditingTitle] = useState('');
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+      inputRef.current.select();
+    }
+  }, [isEditing, todo.id]);
+
+  function handleEdit() {
+    setIsEditing(true);
+    setEditingTitle(todo.title);
+  }
+
+  function handleEditChange(e: React.ChangeEvent<HTMLInputElement>) {
```

Comment:
nit: arrow functions is a more common way to create a function for event handler in components, at least you should be consistent